addons/monitoring/lambda/main.go

```diff
@@ -203,24 +211,43 @@ func checkCrons(db *sql.DB, sess *session.Session) (err error) {
 	}
 	cronAlertTimestamp := time.Now().Add(-1 * cronMonitorInterval)
 
-	// Find all cron entries less than cronMonitorInterval old that have errors.
-	rows, err := db.Query("SELECT name, created_at, IFNULL(updated_at, FROM_UNIXTIME(0)) AS updated_at, errors FROM cron_stats WHERE errors IS NOT NULL AND created_at > \"" + cronAlertTimestamp.Format("20060102150405") + "\"")
-	defer rows.Close()
+	// Gather stats about how many runs raised errors since the last check.
+	rows, err := db.Query(`
+		SELECT 
+			name, 
+			COUNT(*) AS occurences_in_last_hour, 
+			COUNT(errors) as errors_in_last_hour, 
+			MAX(updated_at) AS last_updated_at, 
+			SUBSTRING_INDEX( GROUP_CONCAT(errors ORDER BY updated_at DESC SEPARATOR 0x1e), 0x1e, 1 ) AS most_recent_error 
```

Comment:
0x1e (hex 31) is a non-printing character used here just as a delimiter, instead of using something like `,` or `|` which could possibly show up in an error message.  The code here makes one long string out of all the errors reported in the window, divided by the delimiter, then finds the first one.

Reply:
record separator gang